.github/workflows/version.yml

```diff
@@ -0,0 +1,34 @@
+name: Version Packages
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'docs/**'
+
+concurrency: version-or-publish-${{ github.ref }}
+
+jobs:
+  version:
+    name: Prepare Release PR
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+        with:
+          fetch-depth: 0 # To get all tags
+          ref: ${{ github.ref }}
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - name: Create Prepare Release PR
+        uses: changesets/action@001cd79f0a536e733315164543a727bdf2d70aff # v1.5.1
+        with:
+          title: Prepare Release
+          commit: Prepare Release
+          version: npm run version
+          commitMode: github-api
```

Comment:
Is this specifically for signing commits or any other advantages over cli?

Reply:
Yes, according to the Changesets changelog,
> When used with "github-api" value all tags and commits will be attributed to the user whose GITHUB_TOKEN is used, and also signed using GitHub's internal GPG key.